src/Controller/Action/ShowChosenWishlistAction.php

```diff
@@ -0,0 +1,127 @@
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Twig\Environment;
+
+final class ShowChosenWishlistAction
+{
+    private WishlistRepositoryInterface $wishlistRepository;
+
+    private CartContextInterface $cartContext;
+
+    private FormFactoryInterface $formFactory;
+
+    private OrderModifierInterface $orderModifier;
+
+    private EntityManagerInterface $cartManager;
+
+    private FlashBagInterface $flashBag;
+
+    private TranslatorInterface $translator;
+
+    private Environment $twigEnvironment;
+
+    public function __construct(
+        WishlistRepositoryInterface $wishlistRepository,
+        CartContextInterface $cartContext,
+        FormFactoryInterface $formFactory,
+        OrderModifierInterface $orderModifier,
+        EntityManagerInterface $cartManager,
+        FlashBagInterface $flashBag,
+        TranslatorInterface $translator,
+        Environment $twigEnvironment
+    ) {
+        $this->wishlistRepository = $wishlistRepository;
+        $this->cartContext = $cartContext;
+        $this->formFactory = $formFactory;
+        $this->orderModifier = $orderModifier;
+        $this->flashBag = $flashBag;
+        $this->twigEnvironment = $twigEnvironment;
+        $this->cartManager = $cartManager;
+        $this->translator = $translator;
+    }
+
+    public function __invoke(int $wishlistId, Request $request): Response
+    {
+        /** @var WishlistInterface $wishlist */
+        $wishlist = $this->wishlistRepository->find($wishlistId);
+
+        $cart = $this->cartContext->getCart();
+
+        $form = $this->formFactory->create(AddProductsToCartType::class, null, [
+            'cart' => $cart,
+            'wishlist_products' => $wishlist->getWishlistProducts(),
+        ]);
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+
+            if ($this->handleCartItems($form)) {
+                $this->flashBag->add('success', $this->translator->trans('bitbag_sylius_wishlist_plugin.ui.added_to_cart'));
+            } else {
+                $this->flashBag->add('error', $this->translator->trans('bitbag_sylius_wishlist_plugin.ui.increase_quantity'));
+            }
+
+            return new Response(
+                $this->twigEnvironment->render('@BitBagSyliusWishlistPlugin/WishlistDetails/index.html.twig', [
+                    'wishlist' => $wishlist,
+                    'form' => $form->createView(),
+                ])
+            );
+        }
+
+        foreach ($form->getErrors() as $error) {
+            $this->flashBag->add('error', $error->getMessage());
+        }
+
+        return new Response(
+            $this->twigEnvironment->render('@BitBagSyliusWishlistPlugin/WishlistDetails/index.html.twig', [
+                'wishlist' => $wishlist,
+                'form' => $form->createView(),
+            ])
+        );
+    }
```

Comment:
Too many responsibilities here. Check if this could be done via Sylius Resource.

Reply:
Generally this is logic from default wihslist check but after simply upgrade for checing selected wishlists.